### PR TITLE
Feature/[RDB] block indexer

### PR DESCRIPTION
**Pull request by iceseer**
### Description of the Change
Made BlockIndexer refactoring.
Added RocksDB implementation of Indexer.

### CODE REVIEW

File: shared_model/backend/plain/impl/peer.cpp [hunk 1]
```diff
@@ -8,13 +8,17 @@
...
+           const std::optional<interface::types::TLSCertificateTypeView>
+               &tls_certificate)
     : address_(address),
       public_key_hex_(std::move(public_key_hex)),
-      tls_certificate_(tls_certificate) {}
+      tls_certificate_(
+          tls_certificate
+              ? std::optional<
+                    interface::types::TLSCertificateType>{*tls_certificate}
+              : std::nullopt) {}
```
Comment (lebdron) [inline shared_model/backend/plain/impl/peer.cpp:21]: Why? Field has optional type as well

File: irohad/ametsuchi/impl/rocksdb_common.hpp [hunk 2]
```diff
@@ -752,6 +777,111 @@ namespace iroha::ametsuchi {
...
+    std::optional<std::string_view> tx;
+    if constexpr (kOp == kDbOperation::kGet)
+      if (status.assumeValue().ok())
+        tx = common.valueBuffer();
+
+    return tx;
+  }
+
+  /**
+   * Access to transactions statuses
```
Comment (lebdron) [inline irohad/ametsuchi/impl/rocksdb_common.hpp:810]: Does not look correct, please fix

File: irohad/ametsuchi/impl/rocksdb_common.hpp [hunk 3]
```diff
@@ -752,6 +777,111 @@ namespace iroha::ametsuchi {
...
```
Comment (lebdron) [inline irohad/ametsuchi/impl/rocksdb_common.hpp:849]: Please avoid copy-paste and check all the comments

File: irohad/ametsuchi/impl/rocksdb_indexer.hpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AMETSUCHI_ROCKSDB_INDEXER_HPP
+#define AMETSUCHI_ROCKSDB_INDEXER_HPP
+
+#include "ametsuchi/indexer.hpp"
+
+#include "ametsuchi/impl/rocksdb_common.hpp"
```
Comment (lebdron) [inline irohad/ametsuchi/impl/rocksdb_indexer.hpp:11]: Is not required in the header, you can use forward declarations instead

File: test/module/irohad/ametsuchi/rocksdb_indexer_test.cpp [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+  std::string hash_1_ = "hash1";
+  std::string hash_2_ = "hash2";
+  std::string hash_3_ = "hash3";
+
+  uint64_t ts_1_ = 1001;
+  uint64_t ts_2_ = 1002;
+};
+
+TEST_F(RocksDBIndexerTest, SimpleInsertTxByTs) {
```
Comment (lebdron) [inline test/module/irohad/ametsuchi/rocksdb_indexer_test.cpp:79]: Please add test case descriptions

**Review by lebdron — CHANGES_REQUESTED** [review]

**Review by kuvaldini — APPROVED** [review]

File: irohad/ametsuchi/impl/rocksdb_indexer.hpp (hunk 4 above)
Comment (iceseer) [reply irohad/ametsuchi/impl/rocksdb_indexer.hpp:11]: There, the forward declaration is already used ... Header is inserted by mistake

**Review by lebdron — APPROVED** [review]

**Pull request merged**